classes/fsl-eula-unpack.bbclass

```diff
@@ -34,9 +57,39 @@ python fsl_bin_do_unpack() {
         if not os.path.exists(url.localpath):
             bb.debug(1, "Exiting as '%s' cannot be found" % url.basename)
             return
+        found += 1
         bb.note("Handling file '%s' as a Freescale EULA-licensed archive." % url.basename)
         cmd = "sh %s --auto-accept --force" % (url.localpath)
         bb.fetch2.runfetchcmd(cmd, d, quiet=True, workdir=rootdir)
+
+    # Check for two EULAs, one from the layer and one from the package
+    bb.note("Checking LIC_FILES_CHKSUM for Freescale EULA consistency...")
+    if found > 1:
+        bb.warn("The package contains multiple Freescale EULA-licensed archives. The consistency logic may not be able to detect a EULA problem.")
+    layer_license = d.getVar('LIC_FILES_CHKSUM_LAYER')
+    licenses = d.getVar('LIC_FILES_CHKSUM') or ""
+    md5sums = d.getVar('FSL_EULA_FILE_MD5SUMS') or ""
+    found_layer_license = False
+    found_package_license = False
+    for license in licenses.split():
+        try:
+            (method, host, path, user, pswd, parm) = bb.fetch.decodeurl(license)
```

Comment:
I changed the code order so now the layer license is also subjected to this call to decodeurl.